### PR TITLE
Adding restart implementation

### PR DESCRIPTION
**Pull request by prabdeb**
Adding restart implementation for `kdk restart` by using `DockerClient.ContainerRestart`

### CODE REVIEW

File: pkg/kdk/restart.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+func Restart(cfg KdkEnvConfig, logger logrus.Entry) error {
+
+	var containerIds []string
+
+	containers, err := cfg.DockerClient.ContainerList(cfg.Ctx, types.ContainerListOptions{})
+
+	if err != nil {
+		logger.WithField("error", err).Fatal("Failed to list docker containers")
+	}
+	for _, container := range containers {
```
Comment (dcwangmit01) [inline pkg/kdk/restart.go]: We should break this out into its own function.  I've seen this code before.

File: pkg/kdk/restart.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+			}
+			if result, err := prmpt.Run(); err != nil || result == "n" {
+				logger.Error("KDK container restart canceled or invalid input.")
+				return nil
+			}
+			if err := cfg.DockerClient.ContainerRestart(cfg.Ctx, containerId); err != nil {
+				logger.WithField("error", err).Fatal("Failed to restart KDK container")
+			}
+		}
+		logger.Info("KDK destroy complete.")
```
Comment (dcwangmit01) [inline pkg/kdk/restart.go]: MUST: Copy Pasta

File: pkg/kdk/restart.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+			prmpt := prompt.Prompt{
+				Text:     "Continue? [y/n] ",
+				Loop:     true,
+				Validate: prompt.ValidateYorN,
+			}
+			if result, err := prmpt.Run(); err != nil || result == "n" {
+				logger.Error("KDK container restart canceled or invalid input.")
+				return nil
+			}
+			if err := cfg.DockerClient.ContainerRestart(cfg.Ctx, containerId); err != nil {
```
Comment (dcwangmit01) [inline pkg/kdk/restart.go]: I think in the old Python kdk version the restart might have been a snapshot and then run again, which saved state.   I don't think we're doing this here, and I'm not sure if it is what people expect when "restarting".

File: cmd/kdk/restart.go [hunk 4]
```diff
@@ -25,7 +25,9 @@ var restartCmd = &cobra.Command{
 	Short: "Restart the running KDK container (useful for config changes)",
 	Long:  `Restart the running KDK container (useful for config changes)`,
 	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Println("restart called")
+		logger := logrus.New().WithField("command", "restart")
+
+		kdk.restart(CurrentKdkEnvConfig, *logger)
```
Comment (rtluckie) [inline cmd/kdk/restart.go]: should be `kdk.Restart`

**Comment by prabdeb** [comment]
yes @darkyat we can close it as of now, I will work on this later!

**Comment by dcwangmit01** [comment]
No need to ever implement restart.  It is now built into "kdk ssh" which will detect if its down and prompt the user to restart.

We should remove the restart subcommand.

**Pull request closed without merging**